base/compiler/ssair/passes.jl

```diff
@@ -596,17 +596,16 @@ a result of succeeding dead code elimination.
 """
 function sroa_pass!(ir::IRCode)
     compact = IncrementalCompact(ir)
-    defuses = IdDict{Int, Tuple{IdSet{Int}, SSADefUse}}()
+    defuses = nothing # will be initialized once we encounter mutability in order to reduce dynamic allocations
```

Comment:
I just wonder: we may want to have some nice support for this sort of optimization?
Like: `@delay_alloc defuses = IdDict{Int, Tuple{BitSet, SSADefUse}}()`